conan/cli/formatters/graph/graph.py

```diff
@@ -144,3 +144,44 @@ def format_graph_json(result):
     cli_out_write(json_result)
     if graph.error:
         raise graph.error
+
+
+def format_graph_cyclonedx(result):
+    """
+    # creates a CycloneDX JSON according to https://cyclonedx.org/docs/1.4/json/
+    """
+    def licenses(conanfilelic):
+        def entry(id):
+            return {"license": {
+                "id": id
+            }}
+        if conanfilelic is None:
+            return []
+        elif isinstance(conanfilelic, str):
+            return [entry(conanfilelic)]
+        else:
+            return [entry(i) for i in conanfilelic]
+
+    result["graph"].serialize()  # fills ids
+    deps = result["graph"].nodes[1:]  # first node is app itself
+    cyclonedx = {
+        "bomFormat": "CycloneDX",
+        "specVersion": "1.4",
+        "version": 1,
+        "dependencies": [n.id for n in deps],
+        "components": [
+            {
+                "type": "library",
+                "bom-ref": n.id,
+                "purl": n.package_url().to_string(),
+                "licenses": licenses(n.conanfile.license),
+                "name": n.name,
+                "version": n.conanfile.version,
+                "supplier": {
+                    "url": [n.conanfile.homepage]
```

Comment:
homepage - why not `externalReferences` of type `website`? 
see https://cyclonedx.org/use-cases/#external-references

Reply:
What is the advantage of using that over supplier? external references are not part of the BOM: https://cyclonedx.org/docs/1.4/json/#components_items_supplier VS https://cyclonedx.org/docs/1.4/json/#components_items_externalReferences

Reply:
It's a matter of correctness.
One is the URL to the homepage of a supplier,
the other is the URL to the homepage of the component/project.

From what I've seen, `n.conanfile.homepage` contains the project's homepage, not the homepage of an author or supplier.
see https://docs.conan.io/1/reference/conanfile/attributes.html#homepage

I found that `n.conanfile.url` could be the `externalReference` of type `vcs`
see https://docs.conan.io/1/reference/conanfile/attributes.html#url

Reply:
`conanfile.url` points to the GitHub repo of conan-center. Not sure that is useful in this case.

I will use the homepage attribute and move it to `externalReferences`